### ISSUE TITLE
Automatic migrations migrate to the wrong migration

### ISSUE THREAD

**Issue by mpauly-exnaton** [authzed/spicedb-operator#136]
Hi, thanks for providing SpiceDB and the operator!

Using SpiceDB-Operator version `1.1.0`, we have deployed a SpiceDB Cluster with a postgres instance as the datastore. The operator then spins up the cluster, uses version `v1.16.1`, and first launches a pod to run migrations and then launches the pod running the actual SpiceDB instance. 
Without further changes the `migrate` pod migrated our database to the `drop-id-constraints` migration as evident from the the pod's `Command: spicedb migrate drop-id-constraints`.
However, then the actual SpiceDB pod fails its health check and SpiceDB insists that the datastore is not ready, as evident from the log entry `"datastoreReady":false`. This presumably is a symptom of the migrations being outdated - see https://github.com/authzed/spicedb/issues/739.
After manually migrating with a `k exec -i --tty <spicedb-pod> -- spicedb migrate --log-level trace head` an additional migration is performed and the database is migrated to `drop-bigserial-ids`. The pod then starts regularly and its healthchecks pass.

edit: You can find our cluster's spec below
```
apiVersion: authzed.com/v1alpha1
kind: SpiceDBCluster
metadata:
  name: spicedb-authorisation
spec:
  channel: stable 
  version: v1.16.1
  config:
    logLevel: debug
    datastoreEngine: postgres
    replicas: 1
    skipMigrations: false
  secretName: authorisation-secret
```

**Comment by ecordell** [comment]
@mpauly-exnaton Thanks for the report, we'll get that fixed up and released asap!

**Comment by ecordell** [comment]
This will be fixed in `v1.1.1`, which is currently being built.
